app/controllers/categories_controller.rb

```diff
@@ -10,11 +10,16 @@ def show
   end
   def new
     @category=Category.new
+    2.times {@category.products.build}
   end
 
   def create
 
     @category=Category.new(category_params)
+    product_list = params[:category][:products_attributes].values
+    (0...product_list.length).each do |i|
+      @category.products << Product.create(product_list[i])
```

Comment:
vayena timro pani, you should not do Product.create, category save huda nai product aphai save hunuparcha..........